blockchain/blockchain.go

```diff
@@ -1676,6 +1704,93 @@ func (bc *BlockChain) insertChain(chain types.Blocks) (int, []interface{}, []*ty
 	return 0, events, coalescedLogs, nil
 }
 
+// BlockSubscriptionLoop subscribes blocks from a redis server and processes them.
+// This method is only for KES nodes.
+func (bc *BlockChain) BlockSubscriptionLoop(pool *TxPool) {
+	var ch <-chan *redis.Message
+	logger.Info("subscribe blocks from redis cache")
+
+	pubSub, ok := bc.stateCache.TrieDB().TrieNodeCache().(statedb.BlockPubSub)
```

Comment:
Subscribing a "block" from "trie node cache" look confusing. 
How about making `Blockchain` to have `BlockPubSub` interface as its member variable when we create `Blockchain`?

Reply:
As we discussed it through another channel, go with this code and consider other design when it needed.